crates/uv/tests/it/export.rs

```diff
@@ -4618,3 +4618,556 @@ fn export_only_group_and_extra_conflict() -> Result<()> {
+          "type": "library",
+          "bom-ref": "4-sniffio@1.3.1",
+          "name": "sniffio",
+          "version": "1.3.1",
+          "purl": "pkg:pypi/sniffio@1.3.1"
+        }
+      ],
+      "dependencies": [
+        {
+          "ref": "2-anyio@3.7.0",
+          "dependsOn": [
+            "3-idna@3.6",
+            "4-sniffio@1.3.1"
+          ]
+        },
+        {
+          "ref": "3-idna@3.6",
+          "dependsOn": []
+        },
+        {
+          "ref": "1-project@0.1.0",
+          "dependsOn": [
+            "2-anyio@3.7.0"
+          ]
+        },
+        {
+          "ref": "4-sniffio@1.3.1",
+          "dependsOn": []
+        }
+      ]
+    }
+    ----- stderr -----
+    Resolved 4 packages in [TIME]
+    "#);
+
+    Ok(())
+}
+
+#[test]
+fn cyclonedx_export_git_dependency() -> Result<()> {
+    let context = TestContext::new("3.12").with_cyclonedx_filters();
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["flask @ git+https://github.com/pallets/flask.git@2.3.3"]
+
+        [build-system]
+        requires = ["setuptools>=42"]
+        build-backend = "setuptools.build_meta"
+        "#,
+    )?;
+
+    context.lock().assert().success();
+
+    uv_snapshot!(context.filters(), context.export().arg("--format").arg("cyclonedx1.5"), @r#"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    {
+      "bomFormat": "CycloneDX",
+      "specVersion": "1.5",
+      "version": 1,
+      "serialNumber": "[SERIAL_NUMBER]",
+      "metadata": {
+        "timestamp": "[TIMESTAMP]",
+        "tools": [
+          {
+            "vendor": "Astral Software Inc.",
+            "name": "uv",
+            "version": "[VERSION]"
+          }
+        ],
+        "component": {
+          "type": "application",
+          "bom-ref": "1-project@0.1.0",
+          "name": "project",
+          "version": "0.1.0"
+        }
+      },
+      "components": [
+        {
+          "type": "library",
+          "bom-ref": "2-blinker@1.7.0",
+          "name": "blinker",
+          "version": "1.7.0",
+          "purl": "pkg:pypi/blinker@1.7.0"
+        },
+        {
+          "type": "library",
+          "bom-ref": "3-click@8.1.7",
+          "name": "click",
+          "version": "8.1.7",
+          "purl": "pkg:pypi/click@8.1.7"
+        },
+        {
+          "type": "library",
+          "bom-ref": "4-colorama@0.4.6",
+          "name": "colorama",
+          "version": "0.4.6",
+          "purl": "pkg:pypi/colorama@0.4.6"
+        },
+        {
+          "type": "library",
+          "bom-ref": "5-flask@2.3.3",
+          "name": "flask",
+          "version": "2.3.3",
+          "purl": "pkg:generic/flask@2.3.3?download_url=https://github.com/pallets/flask.git?rev=2.3.3#3205b53c7cf69d17fee49cac6b84978175b7dd73"
```

Comment:
broken result. the `download_url` was not properly escaped, so that the hash and the query string are simply to the purl ... 
expected outcome: `pkg:generic/flask@2.3.3?vcs_url=https://github.com/pallets/flask.git%3Frev%3D2.3.3%233205b53c7cf69d17fee49cac6b84978175b7dd73`